### PR TITLE
nyancat: init at 1.5.2

### PR DESCRIPTION
**Pull request by midchildan**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Infinisil** [comment]
As cool as this is, I'd feel bad for [slowing down everybody's `nixos-rebuild`](https://github.com/NixOS/nixpkgs/issues/57477) for this..

How about using [NUR](https://github.com/nix-community/NUR) instead?

**Comment by midchildan** [comment]
If modules are the main source for the slowdown, would it be acceptable if I just remove the module? The module is just for enabling the telnet server, which is an optional component for this package. I'm totally fine with using NUR if this doesn't work out, though.

**Comment by Infinisil** [comment]
Yeah only modules are a slowdown for nixos-rebuild

**Comment by midchildan** [comment]
The module is now removed :)

**Review by knedlsepp — APPROVED** [review]

File: pkgs/tools/misc/nyancat/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "nyancat-${version}";
+  version = "1.5.2";
+
+  src = fetchurl {
+    url = "https://github.com/klange/nyancat/archive/${version}.tar.gz";
```
Comment (Infinisil) [inline pkgs/tools/misc/nyancat/default.nix]: You should use `fetchFromGitHub` instead

File: pkgs/tools/misc/nyancat/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "nyancat-${version}";
```
Comment (Infinisil) [inline pkgs/tools/misc/nyancat/default.nix]: Since recent times, using `pname = "nyancat"` works, which will automatically set `name` too

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by gnidorah** [comment]
> The module is now removed :)

Would you like to add module to NUR then?

**Comment by Infinisil** [comment]
I mean, not putting the module in nixpkgs due to the current performance problems is just my opinion. If a lot of people want the module in nixpkgs we can do that too.

**Comment by gnidorah** [comment]
> I mean, not putting the module in nixpkgs due to the current performance problems is just my opinion. If a lot of people want the module in nixpkgs we can do that too.

This service is more fun than something practical. I would be ok with having it only via nur

File: pkgs/tools/misc/nyancat/default.nix (hunk 2 above)
Comment (midchildan) [reply pkgs/tools/misc/nyancat/default.nix]: Ah, that's nice.

**Comment by midchildan** [comment]
> Would you like to add module to NUR then?

Not really, since I don't plan to use the telnet functionality myself. I initially included the module because other distributions like Arch and Debian did the same.

**Comment by Infinisil** [comment]
@GrahamcOfBorg build nyancat

**Review by infinisil — APPROVED** [review]

**Pull request merged**